query/src/frontend/influxrpc.rs

```diff
@@ -488,6 +490,60 @@ impl InfluxRPCPlanner {
         Ok(field_list_plan)
     }
 
+    /// Returns a plan that finds all rows rows which pass the
+    /// conditions specified by `predicate` in the form of logical
+    /// time series.
+    ///
+    /// A time series is defined by the unique values in a set of
+    /// "tag_columns" for each field in the "field_columns", ordered by
+    /// the time column.
+    ///
+    /// The output looks like:
+    /// ```text
+    /// (tag_col1, tag_col2, ... field1, field2, ... timestamp)
+    /// ```
+    ///
+    /// The  tag_columns are ordered by name.
+    ///
+    /// The data is sorted on tag_col1, tag_col2, ...) so that all
+    /// rows for a particular series (groups where all tags are the
+    /// same) occur together in the plan
+
+    pub async fn read_filter<D>(&self, database: &D, predicate: Predicate) -> Result<SeriesSetPlans>
+    where
+        D: Database + 'static,
+    {
+        debug!(predicate=?predicate, "planning read_filter");
+
+        // group tables by chunk, pruning if possible
+        // key is table name, values are chunks
+        let mut table_chunks = BTreeMap::new();
+
+        for chunk in self.filtered_chunks(database, &predicate).await? {
+            let table_names = self.chunk_table_names(chunk.as_ref(), &predicate).await?;
+            for table_name in table_names.into_iter() {
+                table_chunks
+                    .entry(table_name)
+                    .or_insert_with(Vec::new)
+                    .push(Arc::clone(&chunk));
+            }
+        }
+
+        // now, build up plans for each table
+        let mut ss_plans = vec![];
```

Comment:
This one looks like we could do:

```suggestion
        let mut ss_plans = Vec::with_capacity(table_chunks.len());
```